tests/integration/test_utils/data_source_config/base.py

```diff
@@ -107,6 +109,33 @@ def setup(self) -> None: ...
     @abstractmethod
     def teardown(self) -> None: ...
 
+    @contextmanager
+    def data_context_test_context(self) -> Generator[AbstractDataContext, None, None]:
+        """Receive a DataContext and ensure proper setup and teardown regardless of errors."""
+        try:
+            self.setup()
+            yield self.context
+        finally:
+            self.teardown()
+
+    @contextmanager
+    def asset_test_context(self) -> Generator[_AssetT, None, None]:
```

Comment:
Bummer to have to have to create these wrappers for all the methods (I can see us also creating them for at least data sources, but probably checkpoints etc later on too), but not the end of the world.

Reply:
yeah. I figured having the `context` one would be a sort of escape path that would allow most of what we need. We only need the `batch` version for this PR, but i figured it was useful to add the others since a) i already find them useful in manual testing/debugging, and b) that way the pattern for extension is clear